src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -0,0 +1,608 @@
+*/
+extern "C" int32_t EnumCalendarInfo(
+	EnumCalendarInfoCallback callback,
+	const UChar* localeName,
+	CalendarId calendarId,
+	CalendarDataType dataType,
+	const void* context)
+{
+	Locale locale = GetLocale(localeName);
+	if (locale.isBogus())
+		return false;
+
+	switch (dataType)
+	{
+		case ShortDates:
+			// ShortDates to map kShort and kMedium in ICU, but also adding the "yMd" skeleton as well, as this
+			// closely matches what is used on Windows
+			return InvokeCallbackForDateTimePattern(locale, UDAT_YEAR_NUM_MONTH_DAY, callback, context) &&
+				InvokeCallbackForDatePattern(locale, DateFormat::kShort, callback, context) &&
+				InvokeCallbackForDatePattern(locale, DateFormat::kMedium, callback, context);
+		case LongDates:
+			// LongDates map to kFull and kLong in ICU.
+			return InvokeCallbackForDatePattern(locale, DateFormat::kFull, callback, context) &&
+				InvokeCallbackForDatePattern(locale, DateFormat::kLong, callback, context);
+		case YearMonths:
+			return InvokeCallbackForDateTimePattern(locale, UDAT_YEAR_MONTH, callback, context);
+		case DayNames:
+			return EnumWeekdays(locale, calendarId, DateFormatSymbols::STANDALONE, DateFormatSymbols::WIDE, callback, context);
+		case AbbrevDayNames:
+			return EnumWeekdays(locale, calendarId, DateFormatSymbols::STANDALONE, DateFormatSymbols::ABBREVIATED, callback, context);
+		case MonthNames:
+			return EnumMonths(locale, calendarId, DateFormatSymbols::STANDALONE, DateFormatSymbols::WIDE, callback, context);
+		case AbbrevMonthNames:
+			return EnumMonths(locale, calendarId, DateFormatSymbols::STANDALONE, DateFormatSymbols::ABBREVIATED, callback, context);
+		case SuperShortDayNames:
+			return EnumWeekdays(locale, calendarId, DateFormatSymbols::STANDALONE, DateFormatSymbols::SHORT, callback, context);
+		case MonthGenitiveNames:
+			return EnumMonths(locale, calendarId, DateFormatSymbols::FORMAT, DateFormatSymbols::WIDE, callback, context);
+		case AbbrevMonthGenitiveNames:
+			return EnumMonths(locale, calendarId, DateFormatSymbols::FORMAT, DateFormatSymbols::ABBREVIATED, callback, context);
+		case EraNames:
+		case AbbrevEraNames:
+			return EnumEraNames(locale, calendarId, dataType, callback, context);
+		default:
+			assert(false);
+			return false;
+	}
+}
+
+/*
+Function:
+GetLatestJapaneseEra
+
+Gets the latest era in the Japanese calendar.
+*/
+extern "C" int32_t GetLatestJapaneseEra()
+{
+	UErrorCode err = U_ZERO_ERROR;
+	Locale japaneseLocale("ja_JP@calendar=japanese");
+	LocalPointer<Calendar> calendar(Calendar::createInstance(japaneseLocale, err));
+
+	if (U_FAILURE(err))
+		return 0;
+
+	return calendar->getMaximum(UCAL_ERA);
+}
+
+/*
+Function:
+GetJapaneseEraInfo
+
+Gets the starting Gregorian date of the specified Japanese Era.
+*/
+extern "C" int32_t GetJapaneseEraStartDate(
+	int32_t era,
+	int32_t* startYear,
+	int32_t* startMonth,
+	int32_t* startDay)
+{
+	UErrorCode err = U_ZERO_ERROR;
+	Locale japaneseLocale("ja_JP@calendar=japanese");
```

Comment:
This string is used in a few places.  Should we separate it out into a single named location?